opencog/atoms/core/Variables.h

```diff
@@ -132,6 +132,11 @@ struct Variables : public FreeVariables
 	VariableDeepTypeMap _deep_typemap;
 	VariableDeepTypeMap _fuzzy_typemap;
 
+	/// Return true iff all variables are well typed. For now only
+	/// simple types are supported, specifically if some variable is
+	/// simple typed NOTYPE, then it returns false.
```

Comment:
what does it mean for something to be "well-typed"? Can you give a definition or a few examples?